pkg/services/accesscontrol/dualwrite/reconciler.go

```diff
@@ -88,6 +100,37 @@ func (z *ZanzanaSynchroniser) Sync(ctx context.Context) error {
 	return nil
 }
 
+// Reconcile schedules as job that will run and reconcile resources between
+// legacy access control and zanzana.
+func (r *ZanzanaReconciler) Reconcile(ctx context.Context) error {
+	// Trigger full reconciliation initially
+	r.reconcile(ctx)
+
+	// FIXME:
+	// 1. We should be a bit graceful about reconciliations so we are not hammering dbs
+	// 2. We should be able to configure reconciliation interval
+	// 3. We should be able to detect if we already have a reconciliation loop running in case they take a long time
```

Comment:
I think we can add a flag `ZanzanaReconciler`, like `inProgress bool` and use mutex to set/release a flag. Then, `reconcile()` func can check if there's any other reconcile in progress and return immediately.

Reply:
Yeah or just an atomic would do fine

Reply:
oh and we need to handle multiple instances so we should probably use serverlock